src/MicrosoftTeams.ts

```diff
@@ -421,22 +422,61 @@ namespace microsoftTeams
             let link = document.createElement("a");
             link.href = authenticateParameters.url;
 
-            let messageId = sendMessage(
-                "authentication.authenticate",
-                link.href,
-                authenticateParameters.width,
-                authenticateParameters.height);
-            callbacks[messageId] = (success: boolean, response: string) =>
+            // Clear any previous success/failure values
+            localStorage.removeItem("authentication.success");
+            localStorage.removeItem("authentication.failure");
+
+            // Open a popup window to the specified authentication URL
+            let authWindow: Window;
+            if (loadedInDesktopClient)
             {
-                if (success)
+                let messageId = sendMessage("authentication.authenticate", link.href);
+                callbacks[messageId] = (allowed: boolean) =>
```

Comment:
We will never even make this call until the client starts returning isDesktopClient as part of the initialization. I will make all the necessary FE client changes in a single check-in to get the desktop client auth flow work end-to-end with this new version of the SDK.